packages/eslint-plugin/tests/rules/restrict-template-expressions.test.ts

```diff
@@ -344,6 +344,52 @@ ruleTester.run('restrict-template-expressions', rule, {
     'const msg = `arg = ${undefined}`;',
     'const msg = `arg = ${123}`;',
     "const msg = `arg = ${'abc'}`;",
+    // https://github.com/typescript-eslint/typescript-eslint/issues/11759
+    // allow should check base types
```

Comment:
```suggestion
```

[Style] Nit: we don't generally put these comments in, unless it's really not clear what the tests are doing. Comments aren't enforced by anything to be correct -- and they have a habit of getting out-of-date over time.

The tests you're adding are nice and clear (yay! nicely done!) so I don't think this or the other comments are needed.